lib/practice_exercises.rb

```diff
@@ -1,13 +1,58 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  raise ArgumentError 'Input is not a string' unless string.class == String
+  return true if string == ""
+  string.upcase!
+
+  beg = 0
+  fin = string.length - 1
+
+  while beg < fin do
+    if string[beg] =~ /[a-z0-9]/i && string[fin] =~ /[a-z0-9]/i
+      return false if string[beg] != string[fin]
+      beg += 1
+      fin -= 1
+    else
+      beg += 1 unless string[beg] =~ /[a-z0-9]/i
+      fin -= 1 unless string[fin] =~ /[a-z0-9]/i
+    end
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(m*n) + O(n) for min_by
+# m = length of shortest word, n = number of input strings
+# Not sure if this would be considered O(n) or O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
You have 2 loops one which goes the length of the shortest string, m.  The second goes through each string in the array, n.  So this is O(nm) time complexity.